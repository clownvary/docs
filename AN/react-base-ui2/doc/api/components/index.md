## CollapsePanel 
---
UI component of CollapsePanel

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
children|node|&ensp;|true|child content of CollapsePanel
className|string|''|false|customize class for CollapsePanel section
expandedIcon|string|'icon-chevron-up'|false|expanded icon of CollapsePanel panel
collapsedIcon|string|'icon-chevron-down'|false|collapsed icon of CollapsePanel panel
transition|string|'none'|false|define collapse/expanded transition, just have a transition on attribute of 'height'
title|string|''|false|define what title will be display on collapse panel
summary|string &#124; element|''|false|define what summary will be display on collapse panel
onExpand|func|() => {}|false|a callback function after expanded
onCollapse|func|() => {}|false|a callback function after collapsed