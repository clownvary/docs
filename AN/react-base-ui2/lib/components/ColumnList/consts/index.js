'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _ColumnType = require('./ColumnType');var _ColumnType2 = _interopRequireDefault(_ColumnType);
var _SelectionMode = require('../../../consts/SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);
var _SortOrder = require('../../../consts/SortOrder');var _SortOrder2 = _interopRequireDefault(_SortOrder);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  SortOrder: _SortOrder2.default,
  ColumnType: _ColumnType2.default,
  SelectionMode: SelectionMode };module.exports = exports['default'];