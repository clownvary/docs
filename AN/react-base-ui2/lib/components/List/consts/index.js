'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _ListType = require('./ListType');var _ListType2 = _interopRequireDefault(_ListType);
var _SelectionMode = require('../../../consts/SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  ListType: _ListType2.default,
  SelectionMode: SelectionMode };module.exports = exports['default'];