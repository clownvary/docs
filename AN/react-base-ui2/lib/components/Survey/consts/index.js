'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _QuestionType = require('./QuestionType');var QuestionType = _interopRequireWildcard(_QuestionType);
var _InputType = require('./InputType');var InputType = _interopRequireWildcard(_InputType);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}exports.default =

{
  QuestionType: QuestionType,
  InputType: InputType };module.exports = exports['default'];