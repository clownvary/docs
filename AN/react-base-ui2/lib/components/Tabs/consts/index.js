'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _TabsClasses = require('./TabsClasses');var _TabsClasses2 = _interopRequireDefault(_TabsClasses);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =


{
  TabsClasses: _TabsClasses2.default };module.exports = exports['default'];