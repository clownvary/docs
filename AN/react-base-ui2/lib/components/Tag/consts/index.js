'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _TagSize = require('./TagSize');var _TagSize2 = _interopRequireDefault(_TagSize);
var _TagType = require('./TagType');var _TagType2 = _interopRequireDefault(_TagType);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  TagSize: _TagSize2.default,
  TagType: _TagType2.default };module.exports = exports['default'];