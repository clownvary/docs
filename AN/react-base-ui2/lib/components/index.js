'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _Breadcrumb = require('./Breadcrumb/');var _Breadcrumb2 = _interopRequireDefault(_Breadcrumb);
var _Calendar = require('./Calendar/');var _Calendar2 = _interopRequireDefault(_Calendar);
var _InputMask = require('./InputMask');var _InputMask2 = _interopRequireDefault(_InputMask);
var _InputMoment = require('./InputMoment');var _InputMoment2 = _interopRequireDefault(_InputMoment);
var _InputDate = require('./InputDate');var _InputDate2 = _interopRequireDefault(_InputDate);
var _InputTime = require('./InputTime');var _InputTime2 = _interopRequireDefault(_InputTime);
var _InputTimeRange = require('./InputTimeRange');var _InputTimeRange2 = _interopRequireDefault(_InputTimeRange);
var _InputDateTime = require('./InputDateTime');var _InputDateTime2 = _interopRequireDefault(_InputDateTime);
var _InputNumeric = require('./InputNumeric');var _InputNumeric2 = _interopRequireDefault(_InputNumeric);
var _List = require('./List');var _List2 = _interopRequireDefault(_List);
var _DatePicker = require('./DatePicker');var _DatePicker2 = _interopRequireDefault(_DatePicker);
var _CollapsePanel = require('./CollapsePanel');var _CollapsePanel2 = _interopRequireDefault(_CollapsePanel);
var _LoadingBar = require('./LoadingBar');var _LoadingBar2 = _interopRequireDefault(_LoadingBar);
var _PropertyList = require('./PropertyList');var _PropertyList2 = _interopRequireDefault(_PropertyList);
var _ComboBox = require('./ComboBox');var _ComboBox2 = _interopRequireDefault(_ComboBox);
var _Checkbox = require('./Checkbox');var _Checkbox2 = _interopRequireDefault(_Checkbox);
var _Radio = require('./Radio');var _Radio2 = _interopRequireDefault(_Radio);
var _Input = require('./Input');var _Input2 = _interopRequireDefault(_Input);
var _Button = require('./Button');var _Button2 = _interopRequireDefault(_Button);
var _Modal = require('./Modal');var _Modal2 = _interopRequireDefault(_Modal);
var _Popover = require('./Popover');var _Popover2 = _interopRequireDefault(_Popover);
var _TextArea = require('./TextArea');var _TextArea2 = _interopRequireDefault(_TextArea);
var _AsyncContent = require('./AsyncContent');var _AsyncContent2 = _interopRequireDefault(_AsyncContent);
var _Progress = require('./Progress');var _Progress2 = _interopRequireDefault(_Progress);
var _FileUpload = require('./FileUpload');var _FileUpload2 = _interopRequireDefault(_FileUpload);
var _Survey = require('./Survey');var _Survey2 = _interopRequireDefault(_Survey);
var _ButtonBar = require('./ButtonBar');var _ButtonBar2 = _interopRequireDefault(_ButtonBar);
var _Tabs = require('./Tabs');var _Tabs2 = _interopRequireDefault(_Tabs);
var _Select = require('./Select');var _Select2 = _interopRequireDefault(_Select);
var _Tag = require('./Tag');var _Tag2 = _interopRequireDefault(_Tag);
var _BackTop = require('./BackTop');var _BackTop2 = _interopRequireDefault(_BackTop);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  Breadcrumb: _Breadcrumb2.default,
  Calendar: _Calendar2.default,
  InputMask: _InputMask2.default,
  InputMoment: _InputMoment2.default,
  InputDate: _InputDate2.default,
  InputTime: _InputTime2.default,
  InputTimeRange: _InputTimeRange2.default,
  InputDateTime: _InputDateTime2.default,
  InputNumeric: _InputNumeric2.default,
  List: _List2.default,
  DatePicker: _DatePicker2.default,
  CollapsePanel: _CollapsePanel2.default,
  LoadingBar: _LoadingBar2.default,
  PropertyList: _PropertyList2.default,
  ComboBox: _ComboBox2.default,
  CheckBox: _Checkbox2.default,
  CheckboxGroup: _Checkbox.CheckboxGroup,
  Radio: _Radio2.default,
  RadioGroup: _Radio.RadioGroup,
  Input: _Input2.default,
  Button: _Button2.default,
  Modal: _Modal2.default,
  Popover: _Popover2.default,
  TextArea: _TextArea2.default,
  AsyncContent: _AsyncContent2.default,
  Progress: _Progress2.default,
  FileUpload: _FileUpload2.default,
  Survey: _Survey2.default,
  ButtonBar: _ButtonBar2.default,
  Tabs: _Tabs2.default,
  Select: _Select2.default,
  Tag: _Tag2.default,
  BackTop: _BackTop2.default };module.exports = exports['default'];