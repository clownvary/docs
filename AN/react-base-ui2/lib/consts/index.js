'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.Size5 = exports.Size3 = exports.Size = exports.CustomAttr = exports.Effect = exports.Browser = exports.OS = exports.KeyCode = exports.Theme = exports.Trigger = exports.SortOrder = exports.SelectionMode = exports.DefaultCSSPrefix = exports.VAlignment = exports.HAlignment = exports.Dock = undefined;var _Dock = require('./Dock');var Dock = _interopRequireWildcard(_Dock);
var _HAlignment = require('./HAlignment');var HAlignment = _interopRequireWildcard(_HAlignment);
var _VAlignment = require('./VAlignment');var VAlignment = _interopRequireWildcard(_VAlignment);
var _DefaultCSSPrefix = require('./DefaultCSSPrefix');var _DefaultCSSPrefix2 = _interopRequireDefault(_DefaultCSSPrefix);
var _SelectionMode = require('./SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);
var _SortOrder = require('./SortOrder');var _SortOrder2 = _interopRequireDefault(_SortOrder);
var _Trigger = require('./Trigger');var Trigger = _interopRequireWildcard(_Trigger);
var _Theme = require('./Theme');var Theme = _interopRequireWildcard(_Theme);
var _KeyCode = require('./KeyCode');var KeyCode = _interopRequireWildcard(_KeyCode);
var _OS = require('./OS');var OS = _interopRequireWildcard(_OS);
var _Browser = require('./Browser');var Browser = _interopRequireWildcard(_Browser);
var _Effect = require('./Effect');var Effect = _interopRequireWildcard(_Effect);
var _CustomAttr = require('./CustomAttr');var CustomAttr = _interopRequireWildcard(_CustomAttr);
var _Size = require('./Size');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}exports.


Dock = Dock;exports.
HAlignment = HAlignment;exports.
VAlignment = VAlignment;exports.
DefaultCSSPrefix = _DefaultCSSPrefix2.default;exports.
SelectionMode = SelectionMode;exports.
SortOrder = _SortOrder2.default;exports.
Trigger = Trigger;exports.
Theme = Theme;exports.
KeyCode = KeyCode;exports.
OS = OS;exports.
Browser = Browser;exports.
Effect = Effect;exports.
CustomAttr = CustomAttr;exports.
Size = _Size.Size;exports.
Size3 = _Size.Size3;exports.
Size5 = _Size.Size5;