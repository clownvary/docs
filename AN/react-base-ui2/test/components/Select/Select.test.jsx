import React from 'react';
import renderer from 'react-test-renderer';
import { mount, ReactWrapper } from 'enzyme';
import { Select } from 'src/components';
import { KeyCode } from 'src/consts';

describe('components/Select', () => {
  const optionData = [
    { text: 'option 1', value: 'v1' },
    { text: 'option 2', value: 'v2' },
    { text: 'option 3', value: 'v3' }
  ];

  const testProps = {
    optionData,
    prefixCls: 'test-ms',
    placeholder: 'test search...'
  };

  it('component renders fine', () => {
    const props = testProps;
    const snapshot = renderer.create(<Select {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('component renders fine with optionMenuRenderer', () => {
    const props = {
      ...testProps,
      optionMenuRenderer: menu => (<div className="customize-menu">{menu}</div>)
    };
    const snapshot = renderer.create(<Select {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('component works fine', () => {
    jest.useFakeTimers();
    const onSelect = jest.fn();
    const onChange = jest.fn();
    const onOptionItemClick = jest.fn();
    const props = {
      ...testProps,
      onSelect,
      onChange,
      onOptionItemClick
    };
    const component = mount(<Select {...props} />);
    const instance = component.instance();

    const outerRenderer = component.find('.test-ms');
    const input = component.find('input');
    const selectRenderer = component.find('.test-ms-renderer');
    const placeholder = component.find('.test-ms-placeholder');

    instance.focus(false);
    expect(instance._focused).toBeFalsy();

    placeholder.simulate('click');
    expect(instance._focused).toBeTruthy();

    input.simulate('focus');
    expect(selectRenderer.hasClass('test-ms-renderer__focused')).toBeTruthy();
    input.simulate('blur');
    jest.runOnlyPendingTimers();
    expect(selectRenderer.hasClass('test-ms-renderer__focused')).toBeFalsy();
    expect(instance._focused).toBeFalsy();

    outerRenderer.simulate('click');
    expect(instance._focused).toBeTruthy();
    expect(component.state().open).toBeTruthy();

    input.simulate('change', { target: { value: '2' } });
    expect(component.state().inputValue).toEqual('2');
    expect(document.querySelectorAll('.test-ms-option-item')).toHaveLength(1);

    input.simulate('keydown', { keyCode: KeyCode.BACKSPACE });
    input.simulate('change', { target: { value: '' } });
    expect(component.state().inputValue).toEqual('');
    expect(document.querySelectorAll('.test-ms-option-item')).toHaveLength(3);

    input.simulate('keydown', { keyCode: KeyCode.ENTER });
    expect(onSelect).toHaveBeenCalled();
    expect(onChange).toHaveBeenCalled();

    input.simulate('keydown', { keyCode: KeyCode.ESCAPE });
    input.simulate('keydown', { keyCode: KeyCode.ESCAPE });
    expect(component.state().open).toBeFalsy();
    expect(component.state().choiceValues).toEqual(['v1']);

    input.simulate('keydown', { keyCode: KeyCode.DOWN });
    input.simulate('keydown', { keyCode: KeyCode.DOWN });
    expect(component.state().open).toBeTruthy();

    component.find('.icon-close').simulate('mousedown');
    expect(component.state().choiceValues).toHaveLength(0);

    new ReactWrapper(instance._selectionMenuRef, true).find('.test-ms-option-item').at(2).simulate('mousedown');
    expect(onOptionItemClick).toHaveBeenCalled();
    expect(component.state().choiceValues).toEqual(['v3']);

    input.simulate('keydown', { keyCode: KeyCode.BACKSPACE });
    expect(component.state().choiceValues).toHaveLength(0);

    input.simulate('keydown', { keyCode: KeyCode.BACKSPACE });
    expect(component.state().choiceValues).toHaveLength(0);
  });

  it('component works fine with choice values setup', () => {
    const props = {
      ...testProps,
      choiceValues: ['v2', 'v3']
    };
    const component = mount(<Select {...props} />);
    expect(component.state().choiceValues).toEqual(['v2', 'v3']);

    const nextChoiceValues = ['v1', 'v2', 'v3'];
    component.setProps({ choiceValues: ['v2', 'v3'] });
    component.setProps({ choiceValues: nextChoiceValues });
    expect(component.state().choiceValues).toEqual(['v1', 'v2', 'v3']);

    component.find('.icon-close').at(0).simulate('mousedown');
  });

  it('component works fine if optionMenuRenderer is null', () => {
    const props = {
      ...testProps,
      optionMenuRenderer: null
    };
    const component = mount(<Select {...props} />);
    const instance = component.instance();

    const outerRenderer = component.find('.test-ms');
    outerRenderer.simulate('click');
    expect(instance._focused).toBeTruthy();
    expect(component.state().open).toBeTruthy();
  });

  it('if creatable is true component works fine', () => {
    const props = {
      ...testProps,
      creatable: true,
      onChange: jest.fn()
    };
    const component = mount(<Select {...props} />);
    const instance = component.instance();

    const outerRenderer = component.find('.test-ms');
    outerRenderer.simulate('click');
    expect(instance._focused).toBeTruthy();
    expect(component.state().open).toBeTruthy();
    new ReactWrapper(instance._selectionMenuRef, true).find('.test-ms-option-item').at(2).simulate('mousedown');
    expect(component.state().open).toBeFalsy;
    expect(component.state().inputValue).toEqual('v3');
    expect(props.onChange).toHaveBeenCalled();
    component.setProps({ inputValue: '' });
    expect(component.state().inputValue).toEqual('');
  });
});
