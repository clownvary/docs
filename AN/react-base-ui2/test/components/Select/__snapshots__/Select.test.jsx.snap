// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Select component renders fine 1`] = `
<div
  className="test-ms"
  onClick={[Function]}
>
  <div
    className="test-ms-renderer test-ms-renderer__empty"
  >
    <div
      className="test-ms-selection"
    >
      <ul>
        <li
          className="test-ms-search"
        >
          <div
            className="test-ms-search__container"
          >
            <input
              aria-autocomplete="list"
              aria-label={undefined}
              aria-labelledby={undefined}
              autoComplete={false}
              className="test-ms-search__field"
              disabled={undefined}
              maxLength={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <span
              className="test-ms-search__field-mirror"
            >
              
               
            </span>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="test-ms-placeholder"
      onClick={[Function]}
    >
      test search...
    </div>
  </div>
</div>
`;

exports[`components/Select component renders fine with optionMenuRenderer 1`] = `
<div
  className="test-ms"
  onClick={[Function]}
>
  <div
    className="test-ms-renderer test-ms-renderer__empty"
  >
    <div
      className="test-ms-selection"
    >
      <ul>
        <li
          className="test-ms-search"
        >
          <div
            className="test-ms-search__container"
          >
            <input
              aria-autocomplete="list"
              aria-label={undefined}
              aria-labelledby={undefined}
              autoComplete={false}
              className="test-ms-search__field"
              disabled={undefined}
              maxLength={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <span
              className="test-ms-search__field-mirror"
            >
              
               
            </span>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="test-ms-placeholder"
      onClick={[Function]}
    >
      test search...
    </div>
  </div>
</div>
`;
